web/ui/src/common/jellyTable.js

```diff
@@ -123,65 +125,89 @@
                         }
                     };
 
-                    getData = function($defer, params) {
-                        var unorderedData = data(),
-                            orderedData;
+                    getData = function ($defer, params) {
+                        var allItems = data(),
+                            totalItemCount = 0,
+                            sortedItems,
+                            tableEntries;
 
-                        // if unorderedData is an object, convert to array
-                        // NOTE: angular.isObject does not consider null to be an object
-                        if(!angular.isArray(unorderedData) && angular.isObject(unorderedData)){
-                            unorderedData = utils.mapToArr(unorderedData);
+                        if (allItems === undefined) {
 
-                        // if it's null, create empty array
-                        } else if(unorderedData === null){
-                            unorderedData = [];
-                        }
+                            // if just intitalized, show loading and default to empty array
+                            allItems = [];
+                            toggleNoData(true);
```

Comment:
`toggleNoData(false)` should be used here. If we have *no* result (aka, undefined), then that doesnt mean there is no data, that means that the data hasnt loaded yet. no data specifically means, we got a result, but the result is an empty list.